src/domain/noteSettings.repository.interface.ts

```diff
@@ -0,0 +1,15 @@
+import type NotesSettings from '@/domain/entities/NotesSettings';
+
+/**
+ * Repository interface describes the methods that required by domain for its business logic implementation
+ */
+export default interface NoteSettingsRepositoryInterface {
+
+  /**
+   * Returns NotesSettings by  id
+   *
+   * @param  id - note  id
+   * @returns NotesSettings | null - NotesSettings instance
+   */
+  getNotesSettingsById(id: string): Promise<NotesSettings | null>;
```

Comment:
```suggestion
  getNotesSettingsById(id: NoteId): Promise<NotesSettings | null>;
```

Reply:
I think it should not return `null`. If an incorrect id will be passed, there will be 404 exception thrown